src/api/cli/create_rawtx.go

```diff
@@ -430,36 +424,39 @@ func getKeys(wlt *wallet.Wallet, outs []UnspentOut) ([]cipher.SecKey, error) {
 	return keys, nil
 }
 
-func getSufficientUnspents(unspents []UnspentOut, coins uint64) ([]UnspentOut, error) {
-	var totalCoins uint64
-	var outs []UnspentOut
+func getSufficientUnspents(unspents *webrpc.OutputsResult, coins uint64) ([]UnspentOut, error) {
+	// get spendable outputs, which are all confirmed outputs without
+	// the spending outputs that are in unconfirmed tx pool.
+	spendableOuts := unspents.Outputs.SpendableOutputs()
+	var spendableCoins uint64
+	var spendOuts []UnspentOut
+	for i, out := range spendableOuts {
+		c, err := droplet.FromString(out.Coins)
+		if err != nil {
+			return nil, err
+		}
 
-	addrOuts := make(map[string][]UnspentOut)
-	for _, u := range unspents {
-		addrOuts[u.Address] = append(addrOuts[u.Address], u)
-	}
+		spendableCoins += c
+		spendOuts = append(spendOuts, UnspentOut{spendableOuts[i]})
 
-	for _, us := range addrOuts {
-		for i, u := range us {
-			c, err := droplet.FromString(u.Coins)
-			if err != nil {
-				return nil, err
-			}
-
-			if c == 0 {
-				continue
-			}
+		if spendableCoins >= coins {
+			return spendOuts, nil
+		}
+	}
 
-			totalCoins += c
-			outs = append(outs, us[i])
+	// get unconfirmed incoming outputs
+	uncfmIncomingOuts := unspents.Outputs.IncomingOutputs
+	uncfmIncoming, err := uncfmIncomingOuts.Balance()
+	if err != nil {
+		return nil, fmt.Errorf("get unconfirmed balance failed: %v", err)
+	}
 
-			if totalCoins >= coins {
-				return outs, nil
-			}
-		}
+	if spendableCoins+uncfmIncoming.Coins < coins {
```

Comment:
Are you sure these two errors are not in the wrong order?

Reply:
Yes, if the spendable coins plus the unconfirmed incoming coins still less than the coins you want to spend, that means you don't have enough coins in wallet.